logrotate.c

```diff
@@ -1221,6 +1226,28 @@ static int copyTruncate(const char *currLog, const char *saveLog, const struct s
                 goto fail;
             }
         }
+#if defined(WITH_FALLOCATE)
+    } else if (flags & LOG_FLAG_COPYREDUCE) {
```

Comment:
I find it counter-intuitive that, if both `copytruncate` and `copyreduce` are set, then `copytruncate` silently takes precedence.

Reply:
ok. I'll add a check to display an error if both copytruncate and copyreduce are specified.